helix-view/src/document.rs

```diff
@@ -1065,7 +1065,21 @@ impl Document {
             let write_result: anyhow::Result<_> = async {
                 let mut dst = tokio::fs::File::create(&write_path).await?;
                 to_writer(&mut dst, encoding_with_bom_info, &text).await?;
-                dst.sync_all().await?;
+                // Ignore ENOTSUP/EOPNOTSUPP (Operation not supported) errors from sync_all()
+                // This is known to occur on SMB filesystems on macOS where fsync is not supported
+                if let Err(e) = dst.sync_all().await {
```

Comment:
Alternatively to looking at the `raw_os_error` would it work to check the `io::Error::kind`?

```rust
match dst.sync_all().await {
    Ok(_) => Ok(()),
    Err(e) if e.kind() == io::ErrorKind::Unsupported => Ok(()),
    Err(e) => Err(e.into()),
}
```